include/cpu/tb.h

```diff
@@ -235,4 +237,23 @@ static inline void tb_add_jump(TranslationBlock *tb, int n, TranslationBlock *tb
     }
 }
 
-#endif
+#ifdef CONFIG_SYMBEX
+/* Get the address of the instruction after the one starting at the given address.
+   Useful for variable-length instruction sets. */
+static inline uint64_t tb_get_next_instruction_addr(TranslationBlock *tb, uint64_t addr) {
+    uint64_t ret = 0;
+    int i;
+
+    for (i = 0; i < tb->precise_entries && ret == 0; i++) {
+        if (tb->pc + tb->precise_pcs[i].guest_pc_increment == addr) {
+            ret = addr + tb->precise_pcs[i].guest_inst_size;
+        }
+    }
+
+    assert(ret != 0);
```

Comment:
I wouldn't assert here and instead return 0 size as an error code, in case the given instruction address does not belong to the TB.